File: quizz-ui/src/app/shared/model/CourseDetail.ts
export class CourseDetail {
  constructor(
    public courseId: number = 0,
    public courseCode: string = '',
    public subjects: SubjectCourse[] = [],
  ) {
  }
}


export class SubjectCourse {
  constructor(
    public subjectId: number = 0,
    public subjectName: string = '',
    public subjectIcon: string = '',
    public isExpanded: boolean = false,
    public exams: ExamCourse[] = [],
  ) {
  }
}

export class ExamCourse {
  constructor(
    public testId: number = 0,
    public name: string = '',
    public startDate: string = '',
    public endDate: string = '',
    public time: number = 0,
    public cssClass: string = '',
    public status: string = '',
    public isJoined: boolean = false,
  ) {
  }
}

File: quizz-service/src/main/java/com/app/quizzservice/request/dto/CourseDetail.java
package com.app.quizzservice.request.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
public class CourseDetail {
    Long courseId;
    String courseCode;

    Long testId;
    String name;
    String startDate;
    String endDate;

    Long subjectId;
    String subjectName;
    String subjectIcon;

    Integer duration;
    Boolean isAttempted;
}
